_posts/2015-04-03-april-newsletter.md

```diff
@@ -0,0 +1,476 @@
+
+@tarruda is continuing to refine the new job API. His work has continued in
+[this latest pull request][job-pr]. This might be a breaking change for existing
+plugins that use the existing job API.
+
+This improvement is to make use of function callbacks (for input, output events)
+instead of the previous way of using [autocommands][job-autocmd] for responding
+to events.
+
+In addition to that, there are more job control functions in VimL including
+[`jobwait()`][job-wait] and [`jobclose()`][job-close].
+
+### So Long `long_u`
+
+In an epic crusade against Vim specific integer types that has spanned two
+previous newsletters (June and November), @elmart has finished removing the
+integer type `long_u`. @elmart's changes were broken up into three last pull
+requests ([1][long-1], [2][long-2], [3][long-3]).
+
+This is a step towards
+@tarruda's goal to [remove all project specific types][long-types]. All that
+remains is `char_u` in which there are ~12,300 occurrences currently in use.
+
+### Adding a EndOfBuffer Highlight Group
+
+@mhinz wanted a way to [remove ~ after the file ends][highlight-idea]. He
+proposed a new [`:highlight`][highlight-docs] group, called
+[`EndOfBuffer`][highlight-endofbuffer], to go with the current [list of
+groups][highlight-groups].
+
+His [proposal][highlight-proposal] was first sent as a patch to
+[vim-dev][highlight-vim-dev] where his changes were accepted by Bram and should
+make it into Vim in the future. @mhinz let @fmoralesc prepare the [pull
+request][highlight-pr] to bring in the changes to Neovim. The pull request has
+since been merged into the master branch.
+
+### Garbage Collection Performance
+
+When using Vim with plugins that allocate a large number of objects there are
+times when the garbage collection will kick in and cause Vim to hang due to the
+time spent looking for objects to free.
+
+This issue was [posted][gc-original] to [vim\_dev][gc-vim-dev] by Yasuhiro
+Matsumoto. A [patch][gc-patch] as written by Ariya Mizutani and proposed to the
+mailing list which improves the algorithm for looking through the objects. In
+Yasuhiro's test case, it took unpatched Vim 100 seconds to execute but took just
+0.134 seconds when the patch was applied.
+
+An [issue][gc-issue] by @Shougo was created to discuss this patch being applied
+to Neovim. @oni-link then created a [pull request][gc-pr] to apply the patch
+which was merged shortly after.
+
+### Vimexpect and Example GDB Plugin
+
+@tarruda created a small Vimscript library in a [pull request][expect-pr] that
+mimics Tcl's [Expect][expect-wiki] for Neovim.
+
+The purpose of Expect (in Tcl) is to make it easy to script and automate text
+based applications that don't provide that functionality.
+
+@tarruda provided an example [plugin for GDB][expect-gdb] as a proof of concept.
+The plugin was created to make it easier to develop Neovim and show the
+vimexpect possibilities.
+
+### Options Updates
+
+#### Defaulting to `nocp`
```

Comment:
IMO this title is misleading, as it implies that `nocp` only had its default changed, when in reality a decent amount of code was removed. I think the following would be more accurate:
```
#### Removal of `compatible`
```